src/Build/BackEnd/BuildManager/BuildManager.cs

```diff
@@ -1847,7 +1852,7 @@ private void ExecuteGraphBuildScheduler(GraphBuildSubmission submission)
             if (submission.BuildRequestData.GraphBuildOptions.Build)
             {
                 // Kick off project cache initialization frontloading
-                Task.Run(() => _projectCacheService.InitializePluginsForGraph(projectGraph, _executionCancellationTokenSource.Token));
+                _projectCacheServiceGraphInitialization = Task.Run(() => _projectCacheService.InitializePluginsForGraph(projectGraph, _executionCancellationTokenSource.Token));
```

Comment:
I'm not really sure I like this. Maybe I should remove the `Task.Run` here and move it into `InitializePluginsForGraph` instead and have the `ProjectCacheService` manage the background task during its own disposal. That way `BuildManager` has 1 less thing to worry about...

Reply:
Stupid question:
Why do we need to do Task.Run for initialization at all? I'd imagine the rest depends on it being initialized, right?

Reply:
Yea, this might be an over-optimization. There's stuff that happens before the graph build starts (getting the target list for each node), but `InitializePluginsForGraph` itself does a bunch of fire-and-forget calls so it shouldn't really be blocking for long anyway.

I think you convinced me to just remove the task for simplicity, and we can re-address it if it shows up as a perf problem later (unlikely).